src/test/regress/expected/multi_citus_tools.out

```diff
@@ -629,6 +629,45 @@ SELECT citus_check_connection_to_node('localhost', :worker_2_port);
 (1 row)
 
 \c - - - :master_port
+SELECT * FROM citus_check_cluster_node_health() ORDER BY 1,2,3,4;
+ from_nodename | from_nodeport | to_nodename | to_nodeport | success
+---------------------------------------------------------------------
+ localhost     |         57637 | localhost   |       57637 | t
+ localhost     |         57637 | localhost   |       57638 | t
+ localhost     |         57638 | localhost   |       57637 | t
+ localhost     |         57638 | localhost   |       57638 | t
+(4 rows)
+
+-- test cluster connectivity when we have broken nodes
+SET client_min_messages TO ERROR;
+SET citus.node_connection_timeout TO 10;
+BEGIN;
+INSERT INTO pg_dist_node VALUES
+    (123456789, 123456789, 'localhost', 123456789),
+    (1234567890, 1234567890, 'www.citusdata.com', 5432);
```

Comment:
Cool test, would be good to have NULL values in the hostname/port for extreme testing.

Reply:
Do you think that I should test for NULL values, even though `nodename` and `nodeport` are not NULLABLE?

Can you think of an easy way to check for connections to nodes with NULL nodenames, without adding new entries to `pg_dist_node`?
```
                                  Table "pg_catalog.pg_dist_node"
      Column      |   Type   | Collation | Nullable |                   Default
------------------+----------+-----------+----------+----------------------------------------------
 nodeid           | integer  |           | not null | nextval('pg_dist_node_nodeid_seq'::regclass)
 groupid          | integer  |           | not null | nextval('pg_dist_groupid_seq'::regclass)
 nodename         | text     |           | not null |
 nodeport         | integer  |           | not null | 5432
 noderack         | text     |           | not null | 'default'::text
 hasmetadata      | boolean  |           | not null | false
 isactive         | boolean  |           | not null | true
 noderole         | noderole |           | not null | 'primary'::noderole
 nodecluster      | name     |           | not null | 'default'::name
 metadatasynced   | boolean  |           |          | false
 shouldhaveshards | boolean  |           | not null | true
 ```

Reply:
If we have any NULL values, the defaults are used. Only the `nodename` does not have a default, and should always be supplied.

I could easily drop `5432` nodeport in this example to partially cover what you proposed in this comment.

Reply:
Given that we are in a TX, we can drop NOT NULL constraint:
```SQL
BEGIN;

alter table pg_dist_node alter COLUMN nodename DROP NOT NULL ;

 insert into pg_dist_node(nodename,nodeport) VALUES (NULL, 5432);
INSERT 0 1
Time: 13.709 ms
select * from citus_check_cluster_node_health(); 
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
Time: 2.603 ms
```
which seems to be crashing. So, we should be fixing it and adding the test then :)

Reply:
The fix could be just hard errors for this unexpected case.